### PR TITLE
PAF-130-Added validation for all phone number entries

### PR DESCRIPTION
**Pull request by TemitopeAyokuHO**
## What?
[PAF-130](https://collaboration.homeoffice.gov.uk/jira/browse/PAF-130)
## Why?
Phone number field accepts any text.
## How?
- added validation control to all phone number text fields 
- added error message once an invalid number checked in the validation.js file
## Testing?
Tested manual in the UI and error message & validation work perfect

### CODE REVIEW

File: apps/paf/translations/src/en/validation.json [hunk 1]
```diff
@@ -93,12 +93,16 @@
     "maxlength": "You can't use more than {{maxlength}} characters for your answer"
   },
   "crime-location-phone": {
+    "required": "Enter a valid phone number",
+    "internationalPhoneNumber": "Please enter a vaild phone number",
```
Comment (jamiecarterHO) [inline apps/paf/translations/src/en/validation.json:97]: Typo in 'valid' here and in line 105, 138 and 166.

File: apps/paf/fields/index.js [hunk 2]
```diff
@@ -578,7 +578,7 @@ module.exports = {
   },
   'crime-location-phone': {
     className: ['govuk-input', 'govuk-input--width-20'],
-    validate: [{ type: 'maxlength', arguments: 20 }]
+    validate: ['required', 'internationalPhoneNumber', { type: 'maxlength', arguments: 20 }]
```
Comment (jamiecarterHO) [inline apps/paf/fields/index.js:581]: Is 20 chars enough for a full phone number? Might be worth checking on other forms what they have added here, but probably ok if that has been suggested at some point

**Review by jamiecarterHO — CHANGES_REQUESTED** [review]
Just a typo repeated on a few lines and a question about max char length for phone numbers

File: apps/paf/fields/index.js (hunk 2 above)
Comment (TemitopeAyokuHO) [reply apps/paf/fields/index.js:581]: will 25 do ?

File: apps/paf/fields/index.js [hunk 3]
```diff
@@ -641,7 +641,7 @@ module.exports = {
   },
   'crime-another-location-phone': {
     className: ['govuk-input', 'govuk-input--width-20'],
-    validate: [{ type: 'maxlength', arguments: 20 }]
+    validate: ['required', 'internationalPhoneNumber', { type: 'maxlength', arguments: 20 }]
```
Comment (jamiecarterHO) [inline apps/paf/fields/index.js:644]: Will need to up the number for all of them - lines 581, 644, 792, 876, 936, 1192 and any others.

File: apps/paf/fields/index.js [hunk 4]
```diff
@@ -578,7 +578,7 @@ module.exports = {
   },
   'crime-location-phone': {
     className: ['govuk-input', 'govuk-input--width-20'],
-    validate: [{ type: 'maxlength', arguments: 20 }]
+    validate: ['required', 'internationalPhoneNumber', { type: 'maxlength', arguments: 25 }]
```
Comment (jamiecarterHO) [inline apps/paf/fields/index.js:581]: Other forms (ACQ/SAP, Firearms, ASC) have 200 as the number. Do you think that's too high?

Comment (jamiecarterHO) [reply apps/paf/fields/index.js:581]: How about 50 chars? That would seem to me to be more than enough for most use cases, and cvan always be upped if we get reports of odd cases failing due to not enough chars.

Comment (TemitopeAyokuHO) [reply apps/paf/fields/index.js:581]: Really for phone number ?  but normally phone length is not mor then 30 both for international and local

Comment (jamiecarterHO) [reply apps/paf/fields/index.js:581]: Normally, yes...but it is probably best to cater to potential edge cases. 30 is probably fine too, but I don't see the issue with being at least a little more generous.

File: apps/paf/fields/index.js [hunk 5]
```diff
@@ -1413,7 +1413,7 @@ module.exports = {
       field: 'are-you-eighteen',
       value: 'yes'
     },
-    validate: [{ type: 'maxlength', arguments: 20 }]
+    validate: ['required', 'internationalPhoneNumber', { type: 'maxlength', arguments: 30 }]
```
Comment (jamiecarterHO) [inline apps/paf/fields/index.js:1416]: Is this ok at 30 or should it be 200 like the others? I'd say best to keep them all equal

File: apps/paf/translations/src/en/validation.json [hunk 6]
```diff
@@ -313,7 +316,9 @@
     "after": "Enter a date after 01 01 1900"
   },
   "contact-number": {
-    "maxlength": "You can't use more than {{maxlength}} characters for your answer"
+    "maxlength": "You can't use more than {{maxlength}} characters for your answer",
+    "required": "Please enter a phone number",
```
Comment (jamiecarterHO) [inline apps/paf/translations/src/en/validation.json:320]: This required message is different to the others. They should probably all be the same unless theres a good reason.

**Review by jamiecarterHO — CHANGES_REQUESTED** [review]
Just a couple of values that are different to the others like them

File: apps/paf/fields/index.js (hunk 5 above)
Comment (TemitopeAyokuHO) [reply apps/paf/fields/index.js:1416]: Yes i changed it since other where 200.

**Review by TemitopeAyokuHO — COMMENTED** [review]
replaced the argument size to 200

**Comment by jamiecarterHO** [comment]
Hey @TemitopeAyokuHO really sorry as I asked you to up the max length values for all these phone numbers but it turns out that the IMS max length for phone number fields is 20 - so all those values will need to be 20 in the form as well.

Seems a little short to me still but as it could cause errors otherwise that's what we have to go with!

**Review by TemitopeAyokuHO — COMMENTED** [review]
done

**Pull request merged**